### PR TITLE
add debug info when test failed

### PR DESCRIPTION
**Pull request by songleo**
trace this issue: https://github.com/open-cluster-management/backlog/issues/13901

Signed-off-by: Song Song Li <ssli@redhat.com>

### CODE REVIEW

File: pkg/tests/observability_install_test.go [hunk 1]
```diff
@@ -64,14 +64,13 @@ func installMCO() {
...
 			allPodsIsReady = utils.StatusContainsTypeEqualTo(instance, "Ready")
-			return allPodsIsReady
 		}
-		return false
-	}, EventuallyTimeoutMinute*10, EventuallyIntervalSecond*5).Should(BeTrue())
 
-	if !allPodsIsReady {
-		utils.PrintAllMCOPodsStatus(testOptions)
-	}
+		if !allPodsIsReady {
```
Comment (clyang82) [inline pkg/tests/observability_install_test.go:69]: It can be able to print repeatedly. right?

Comment (songleo) [reply pkg/tests/observability_install_test.go:69]: yes, I think it should be the same with 2.3 https://github.com/open-cluster-management/observability-e2e-test/blob/main/pkg/tests/observability_install_test.go#L146  

```
	By("Waiting for MCO ready status")
	Eventually(func() error {
		err = utils.CheckMCOComponentsInHighMode(testOptions)
		if err != nil {
			testFailed = true
			utils.PrintAllMCOPodsStatus(testOptions)
			return err
		}
		testFailed = false
		return nil
	}, EventuallyTimeoutMinute*25, EventuallyIntervalSecond*10).Should(Succeed())
```

Comment (songleo) [reply pkg/tests/observability_install_test.go:69]: if we put the `utils.PrintAllMCOPodsStatus(testOptions)` outside of `Eventually`.  the test case will return when `Eventually` failed. we can not get the Pod status.

Comment (clyang82) [reply pkg/tests/observability_install_test.go:69]: Why cannot we do it in `afterEach`?

Comment (songleo) [reply pkg/tests/observability_install_test.go:69]: For the install test case, we do not have `beforeEach` and `afterEach`.  we call install case in `BeforeSuite`.

**Pull request closed without merging**